x-pack/platform/plugins/shared/streams_app/public/components/data_management/stream_detail_routing/review_suggestions_form/use_review_suggestions_form.tsx

```diff
@@ -90,6 +90,13 @@ export function useReviewSuggestionsForm() {
     }
   };
 
+  const updateSuggestion = (index: number, updates: Partial<PartitionSuggestion>) => {
+    if (!suggestions) return;
+    const updatedSuggestion = { ...suggestions[index], ...updates };
+    const updatedSuggestions = suggestions.toSpliced(index, 1, updatedSuggestion);
+    setSuggestions(updatedSuggestions);
+  };
+
```

Comment:
This is one way of doing it but it might be easier to manage the list of suggestions using a dedicated hook that simplifies working with arrays.

For example using `useList`: https://streamich.github.io/react-use/?path=/story/state-uselist--docs

```ts
import useList from 'react-use/lib/useList';

const [suggestions, { updateAt, removeAt }] = useList<PartitionSuggestion>()

const removeSuggestion = (index: number) => {
    removeAt(index)
}

const updateSuggestion = (index: number, updates) => {
   updateAt(index, updates);
}
```